mlflow/spark.py

```diff
@@ -58,7 +119,11 @@ def save_model(spark_model, path, mlflow_model=Model(), conda_env=None, jars=Non
             str(type(spark_model))))
     if not isinstance(spark_model, PipelineModel):
         raise Exception("Not a PipelineModel. SparkML can save only PipelineModels.")
-    spark_model.save(os.path.join(path, "model"))
+    # Spark ML stores the model on DFS if running on a cluster
+    # Save it to a DFS temp dir first and copy it to local path
+    tmp_path = _tmp_path(dfs_tmpdir)
+    spark_model.save(tmp_path)
+    _HadoopFileSystem.copy_to_local_file(tmp_path, os.path.abspath(os.path.join(path, "model")))
```

Comment:
We should delete tmp_path, right? I think we only delete it in the `load_model` case.

Reply:
I see, this may be confusing.  The copy_to_local_file call removes the src path. 

Hadoop FileSystem has option to delete src in the copy operations, by default I set it to True  for copy_to_local and to False when doing copy_from_local.

Reply:
Oh... that is subtle. Maybe we can not provide a default and explicitly say removeSrc=True or whatnot inline?